internal/global/trace.go

```diff
@@ -87,11 +87,35 @@ func (p *tracerProvider) Tracer(name string, opts ...trace.TracerOption) trace.T
 		return p.delegate.Tracer(name, opts...)
 	}
 
+	// At this moment it is guaranteed that no sdk is installed, save the tracer in the tracers map.
+
+	key := createKey(name, opts...)
+
+	if p.tracers == nil {
+		p.tracers = make(map[il]*tracer)
+	}
+
+	if val, ok := p.tracers[key]; ok {
+		return val
+	}
+
 	t := &tracer{name: name, opts: opts}
-	p.tracers = append(p.tracers, t)
+	p.tracers[key] = t
 	return t
 }
 
+type il struct {
+	name    string
+	version string
+}
+
+func createKey(name string, opts ...trace.TracerOption) il {
+	return il{
+		name:    name,
+		version: trace.NewTracerConfig(opts...).InstrumentationVersion,
+	}
+}
+
```

Comment:
This function seem superfluous. Can we just move the struct value creation to the one spot this is called?